pipeline-modules/deployment-service/aws/ecs-template.yaml

```diff
@@ -133,9 +137,85 @@ template: |
         - sed '$ s/,$//' containerDef.json > containerDef.json.tmp && mv containerDef.json.tmp containerDef.json
         - echo '],"logConfiguration":'$logConfiguration'}],"requiresCompatibilities":["EC2","FARGATE"],"cpu":"{{container_cpu}}","memory":"{{container_mem}}"}' >> containerDef.json
         - cat containerDef.json
+        - echo "before"
+        - |
+          function update_task_definition_secrets() {
+              task_definition_file="$1"
+              secret_name="$2"
+              secret_arn=$(get_secret_arn "$secret_name")
+              secret_content=$(get_secret_content "$secret_name")
+              secret_content_keys=$(jq --arg secret_arn "$secret_arn" -c '[to_entries[] | {name: .key, valueFrom: ($secret_arn + ":" + .key + "::") }]' <<< "$secret_content")
+              jq --argjson keys "$secret_content_keys" '(.containerDefinitions[] | .secrets) = (.secrets // []) + $keys' "$task_definition_file" > "tmp_task_definition.json" && mv "tmp_task_definition.json" "$task_definition_file"
+          }
+          chmod +x {{ common_scripts_path }}/aws-secret-manager.sh
+          source aws-secret-manager.sh
+          secret_name="{{applicationName}}{{environment}}secrets"
```

Comment:
these names could be anything (could contain special chars and/or 128 chars long). We probably should use our custom filter here.
trimLength and/or replacespecial.

https://webapp.chatgpt4google.com/s/MzA2NTc5